test/src/main/java/org/apache/accumulo/test/ShellServerIT.java

```diff
@@ -1431,18 +1436,20 @@ public void help() throws Exception {
     }
   }
 
-  // TODO - evaluate this test is testing what is expected or history is working.
   @Test
   public void history() throws Exception {
-    final String table = name.getMethodName();
+    final String table = getUniqueNames(1)[0];
 
-    ts.exec("history -c", true);
-    ts.exec("createtable " + table);
-    ts.exec("deletetable -f " + table);
-    // TODO - this may be passing only because method name is history.
+    // test clear history command works
+    ts.writeToHistory("foo");
+    ts.exec("history -c", true, "foo", false);
```

Comment:
This didn't first verify that "foo" was present, or that it was removed afterwards:

```suggestion
    ts.exec("history", true, "foo", true);
    ts.exec("history -c", true);
    ts.exec("history", true, "foo", false);
```

Reply:
Fair point, I was assuming the string present would account for the clear command as well and do the check after clearing. Unsure if that is the case but it was failing when I had the stringPresent set to true

Reply:
I'm pretty sure `history -c` just clears, and doesn't print anything. So, a check for any string present at all would have been false.